package/MDAnalysis/analysis/density.py

```diff
@@ -384,6 +391,19 @@ def __init__(self, atomgroup, delta=1.0,
         self._zdim = zdim
 
     def _prepare(self):
+        r"""Function to prepare Density
+
+        Raises
+        ------
+        ValueError
+            if AtomGroup is empty and no user defined grid is provided
+        UserWarning
+            if AtomGroup is empty and a user defined grid is provided
+
+        .. versionadded:: 1.0.0    
+        .. versionchanged:: 2.0.0
+           Now a method of :class:`DensityAnalysis`.
```

Comment:
Move this doc string into the class doc string. See comment above.

Reply:
Also, remove the `Now a method of :class:DensityAnalysis.` comment as it does not make sense. `_prepare` is always in a AnalysisClass.

Your change is for 2.0.0 so do not add any versionadded for 1.0.0.

Reply:
Thanks again @orbeckst for your patience and advice! My next commit will make these changes..